lib/GHCup/Download.hs

```diff
@@ -124,6 +167,23 @@ getDownloadsF :: ( FromJSONKey Tool
                    GHCupInfo
 getDownloadsF pfreq@(PlatformRequest arch plat _) = do
   Settings { urlSource } <- lift getSettings
+
+  -- Log all URL sources
+  logDebug $ "URL sources in config: " <> T.pack (show urlSource)
+
+  -- Check for custom URL sources that aren't in the default list
+  forM_ urlSource $ \src -> 
+    case src of
+      NewURI uri -> do
+        let url = show uri
+        logDebug $ "Checking URI: " <> T.pack url
+        unless (url `elem` defaultURLs) $ liftIO $ checkUrlPrefix url
```

Comment:
`unless` part is not necessary.

We must show this warning for `https://ghc.gitlab.haskell.org/ghcup-metadata/ghcup-nightlies-0.0.7.yaml`